fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/query/plan/cascades/values/simplification/ValueSimplificationTest.java

```diff
@@ -0,0 +1,417 @@
+
+        final var orderingValues = Values.orderingValuesFromType(type, () -> someCurrentValue, ImmutableSet.of());
+
+        final var expectedResult =
+                ImmutableSet.of(
+                        FieldValue.ofFieldNames(someCurrentValue, ImmutableList.of("a", "aa", "aaa")),
+                        FieldValue.ofFieldNames(someCurrentValue, ImmutableList.of("a", "aa", "aab")),
+                        FieldValue.ofFieldNames(someCurrentValue, ImmutableList.of("a", "aa", "aac")),
+                        FieldValue.ofFieldNames(someCurrentValue, ImmutableList.of("a", "ab")),
+                        FieldValue.ofFieldNames(someCurrentValue, ImmutableList.of("a", "ac")),
+                        FieldValue.ofFieldNames(someCurrentValue, ImmutableList.of("x", "xa")),
+                        FieldValue.ofFieldNames(someCurrentValue, ImmutableList.of("x", "xb")),
+                        FieldValue.ofFieldNames(someCurrentValue, ImmutableList.of("x", "xc")),
+                        FieldValue.ofFieldName(someCurrentValue, "z"));
+
+        Assertions.assertEquals(expectedResult, orderingValues);
+    }
+
+    @Nonnull
+    private static Type.Record someRecordType() {
+        final var aaType = Type.Record.fromFields(ImmutableList.of(
+                Type.Record.Field.of(Type.primitiveType(Type.TypeCode.STRING), Optional.of("aaa")),
+                Type.Record.Field.of(Type.primitiveType(Type.TypeCode.INT), Optional.of("aab")),
+                Type.Record.Field.of(Type.primitiveType(Type.TypeCode.STRING), Optional.of("aac"))));
+
+        final var aType = Type.Record.fromFields(ImmutableList.of(
+                Type.Record.Field.of(aaType, Optional.of("aa")),
+                Type.Record.Field.of(Type.primitiveType(Type.TypeCode.INT), Optional.of("ab")),
+                Type.Record.Field.of(Type.primitiveType(Type.TypeCode.STRING), Optional.of("ac"))));
+
+        final var xType = Type.Record.fromFields(ImmutableList.of(
+                Type.Record.Field.of(Type.primitiveType(Type.TypeCode.STRING), Optional.of("xa")),
+                Type.Record.Field.of(Type.primitiveType(Type.TypeCode.INT), Optional.of("xb")),
+                Type.Record.Field.of(Type.primitiveType(Type.TypeCode.STRING), Optional.of("xc"))));
+
+        return Type.Record.fromFields(ImmutableList.of(
+                Type.Record.Field.of(aType, Optional.of("a")),
+                Type.Record.Field.of(xType, Optional.of("x")),
+                Type.Record.Field.of(Type.primitiveType(Type.TypeCode.STRING), Optional.of("z"))));
+    }
+
+    @Nonnull
+    private static Value defaultSimplify(@Nonnull final Value toBeSimplified) {
+        return toBeSimplified.simplify(DefaultValueSimplificationRuleSet.ofSimplificationRules(), AliasMap.emptyMap(), ImmutableSet.of());
+    }
+
+    @Nonnull
+    private static Value simplifyWithConstantAliases(@Nonnull final Value toBeSimplified, @Nonnull Set<CorrelationIdentifier> constantAliases) {
```

Comment:
Where is this used?

Reply:
Removed. I could swear there is a PMD rule that should have barked.